dnf/rpm/__init__.py

```diff
@@ -48,11 +48,15 @@ def detect_releasever(installroot):
             raise dnf.exceptions.Error(msg)
         releasever = hdr['version']
         try:
-            off = hdr[rpm.RPMTAG_PROVIDENAME].index(distroverpkg)
+            try:
+                # rpm has switched from bytes to strings -> this works only on old rpm versions
```

Comment:
This comment probably should be reworked to be clearer. I'd suggest noting each path in `try: ... except: ...` statement.

Reply:
Reworded the comment, split into both branches.